Core/include/Acts/Utilities/Delegate.hpp

```diff
@@ -209,8 +235,6 @@ class Delegate<R(Args...), H, O> {
             const auto *concretePayload = static_cast<const Type *>(payload);
             delete concretePayload;
           });
-    } else {
-      m_payload.payload = instance.release();
```

Comment:
Sorry @benjaminhuth I just discovered this on top: was this a memory leak before? A non-owning delegate could have a unique_ptr connected, which would release the unique_ptr but never delete it...